docs/docs/01-ibc/06-channel-upgrades.md

```diff
@@ -147,6 +147,52 @@ The application's `OnChanUpgradeRestore` callback method will be invoked.
 
 It will then be possible to re-initiate an upgrade by sending a `MsgChannelOpenInit` message.
 
+## Timing Out a Channel Upgrade
+
+Timing out an outstanding channel upgrade may be necessary during the flushing packet stage of the channel upgrade process. As stated above, with `ChanUpgradeTry` or `ChanUpgradeAck`, the channel state has been changed from `OPEN` to `FLUSHING`, so no new packets will be allowed to be sent over this channel while flushing. However, if flushing exceeds the time limit set in the `UpgradeTimeout` channel `Params`, the upgrade process will need to be timed out.
+
+Timing out an outstanding upgrade is performed by first setting a timeout in the channel `Params` before submitting a `MsgChannelUpgradeTry` or `MsgChannelUpgradeAck` message (by default, 10 minutes): 
+
+```go
+type Params struct {
+	UpgradeTimeout Timeout 
+}
+```
+
+A valid Timeout contains either one or both of a timestamp and block height (sequence).
+
+```go
+type Timeout struct {
+	// block height after which the packet or upgrade times out
+	Height types.Height
+	// block timestamp (in nanoseconds) after which the packet or upgrade times out
+	Timestamp uint64
+}
+```
+
+This timeout will then be set as a field on the `Upgrade` struct itself when flushing is started.
+
+```go
+type Upgrade struct {
+	Fields           UpgradeFields 
+	Timeout          Timeout       
+	NextSequenceSend uint64        
+}
+```
+
+If the timeout has been exceeded during flushing, a chain can then submit the `MsgChannelUpgradeTimeout` to timeout the channel upgrade process and restore the previous channel state:
```

Comment:
I think the upgrade needs to be re-approved by governance if it times out? Can we note that the upgrade process starts from the beginning?